modules/recording/recording/libs/audio-recorder.ts

```diff
@@ -1,66 +1,76 @@
 export class AudioRecorder {
   private mediaRecorder: MediaRecorder | null = null;
   private audioChunks: Blob[] = [];
-  private stream: MediaStream | null = null;
-
-  async startRecording(deviceId?: string): Promise<void> {
-    if (this.mediaRecorder && this.mediaRecorder.state === "paused") {
-      this.mediaRecorder.resume();
-      return;
-    }
+  private audioContext: AudioContext | null = null;
+  private sourceNode: MediaStreamAudioSourceNode | null = null;
 
+  async startRecording(deviceId: string) {
     try {
-      const constraints: MediaStreamConstraints = {
-        audio: deviceId ? { deviceId: { exact: deviceId } } : true,
-      };
-      this.stream = await navigator.mediaDevices.getUserMedia(constraints);
-      this.mediaRecorder = new MediaRecorder(this.stream);
+      const stream = await navigator.mediaDevices.getUserMedia({
+        audio: { deviceId: deviceId ? { exact: deviceId } : undefined },
+      });
+
+      this.audioContext = new (window.AudioContext ||
+        (window as any).webkitAudioContext)();
```

Comment:
`as { webkitAudioContext: () => void }` If we dont know the full type we can extract what we need , can try `satisfies` instead of `as` not sure will work here